### PR TITLE
ci: Add sphinx linkcheck builder to docs build workflow

### LINKED ISSUE #1853: Add linkcheck to docs workflows
With the addition of [user-defined build jobs in ReadTheDocs](https://twitter.com/readthedocs/status/1519363742869295105?s=11&t=5-u_2BFwXLAj9IyXQLhIVA) I noticed that one of their examples was to [perform a check for broken links]( https://docs.readthedocs.io/en/latest/build-customization.html#perform-a-check-for-broken-links) with `sphinx`'s `linkcheck`.

I'm working on adding this both to the ReadTheDocs config and to the docs GHA workflow, but at the moment

```console
$ cd docs
$ make linkcheck
```

is giving a failure

```
(           babel: line    3) broken    cli.html#pyhf-xml2json -
```

on

https://github.com/scikit-hep/pyhf/blob/e7996e5ba350a48825d9736ccc81ca8e3009dd3c/docs/babel.rst?plain=1#L5

I'm not quite sure why, as this is a valid link once the source is built, but I think it might be a form of https://github.com/sphinx-doc/sphinx/issues/9383.

I have this and other fixes on a branch named `docs/use-read-the-docs-pre-build-job`.

### PR DESCRIPTION
**Pull request by matthewfeickert**
# Description

Resolves #1853

* Rename ReadTheDocs config file to `.readthedocs.yaml` as [RTD now requires that exact name](https://docs.readthedocs.io/en/latest/config-file/v2.html).
* Add linkcheck builder config options to Sphinx conf.py to avoid false positives (c.f. #1853).
* Add linkcheck builder runs to docs GitHub Actions workflow and to the ReadTheDocs config
   - Running the linkcheck builder is now possible on ReadTheDocs given [added support for user-defined build jobs](https://docs.readthedocs.io/en/latest/build-customization.html).
* Configure ReadTheDocs workflow to use Python 3.10 and ensure `curl` and `jq` are available.

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Rename ReadTheDocs config file to '.readthedocs.yaml' as RTD now requires that
exact name.
   - c.f. https://docs.readthedocs.io/en/latest/config-file/v2.html
* Add linkcheck builder config options to Sphinx conf.py to avoid false positives.
* Add linkcheck builder runs to docs GitHub Actions workflow and to the ReadTheDocs
config.
   - Running the linkcheck builder is now possible on ReadTheDocs given added support for
   user-defined build jobs. c.f. https://docs.readthedocs.io/en/latest/build-customization.html
* Configure ReadTheDocs workflow to use Python 3.10 and ensure `curl` and `jq` are available.
```

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  os: ubuntu-20.04
+  tools:
+    python: "3.10"
+  apt_packages:
+    - curl
+    - jq
+  # https://docs.readthedocs.io/en/latest/build-customization.html
+  jobs:
+    pre_build:
+      - python -m sphinx -b linkcheck docs/ _build/linkcheck
```
Comment (matthewfeickert) [inline .readthedocs.yaml:19]: I guess a question is should we bother running the linkcheck builder on RTD if we're also running it in the GitHub Actions workflow?

**Comment by matthewfeickert** [comment]
RTD is failing right now with

```
(       citations: line   25) broken    https://doi.org/10.1051/epjconf/202125102070 - Exceeded 30 redirects.
(       citations: line   25) broken    https://doi.org/10.1051/epjconf/202125103067 - Exceeded 30 redirects.
build finished with problems.
```

but this was working earlier (and even passed on GHA). I think that tonight `www.epj-conferences.org` is having server issues.

**Comment by matthewfeickert** [comment]
On this point of DOI links failing on redirects because of servers being weird, @dopplershift makes a point that this [might be more hassle to deal with then it is worth](https://twitter.com/dopplershift/status/1519574223601160196?s=20&t=w-ERpKpzxI5sErjf_zDdSA). Maybe we should extend the list of regexes in `linkcheck_ignore`?

**Comment by dopplershift** [comment]
Looks like I was slightly off. We decided to ban DOIs for [particular problematic journals/publishers](https://github.com/Unidata/MetPy/blob/70d0015000acbb2b865aa33868112306f1f68259/docs/conf.py#L378-L381).

**Review by kratsg — APPROVED** [review]

**Pull request merged**